src/redirect.rs

```diff
@@ -128,7 +128,7 @@ impl Policy {
         match self.inner {
             PolicyKind::Custom(ref custom) => custom(attempt),
             PolicyKind::Limit(max) => {
-                if attempt.previous.len() == max {
+                if attempt.previous.len() >= max {
```

Comment:
Wouldn't this still be without the change? The previous length should be 0, and max is 0... Is there a time that isn't true?

Reply:
Oh, of course! Want to readd the test but with a URL in the previous Vec as well, so we catch this if it ever changes?